src/microbe_stage/MicrobeAI.cs

```diff
@@ -38,13 +39,26 @@ public class MicrobeAI
     [JsonProperty]
     private float pursuitThreshold;
 
+    /// <remarks>
+    ///   Memory of the previous absorption step is required to compute gradient.
+    ///   Values dictionary rather than single value as they will be combined with variable weights.
+    /// </remarks>
+    [JsonProperty]
+    private Dictionary<Compound, float> previouslyAbsorbedCompounds;
+
+    [JsonIgnore]
+    private Dictionary<Compound, float> compoundsSearchWeights;
+
     public MicrobeAI(Microbe microbe)
     {
         this.microbe = microbe ?? throw new ArgumentException("no microbe given", nameof(microbe));
         glucose = SimulationParameters.Instance.GetCompound("glucose");
         oxytoxy = SimulationParameters.Instance.GetCompound("oxytoxy");
         ammonia = SimulationParameters.Instance.GetCompound("ammonia");
         phosphates = SimulationParameters.Instance.GetCompound("phosphates");
+        iron = SimulationParameters.Instance.GetCompound("iron");
```

Comment:
I'd move this line under the glucose definition to match the variable declaration above.